### PR TITLE
MULE-9708 Move all MP notification and exception handling logic to De…

### PR DESCRIPTION
**Pull request by dfeist**
…faultMessageProcessorChain

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/processor/chain/DefaultMessageProcessorChainTestCase.java [hunk 1]
```diff
@@ -317,7 +317,7 @@ public void testMixedMPChainWithVoidReturn1() throws MuleException, Exception
         builder.chain(new AppendingInterceptingMP("1"), new ReturnVoidMPInterceptongMP(),
                       getAppendingMP("2"), getAppendingMP("3"), new AppendingInterceptingMP("4"),
                       getAppendingMP("5"));
-        assertSame("0", process(builder.build(), getTestEventUsingFlow("0")).getMessage().getPayload());
+        assertEquals("0before1after1", process(builder.build(), getTestEventUsingFlow("0")).getMessage().getPayload());
```
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/runtime/core/processor/chain/DefaultMessageProcessorChainTestCase.java]: hamcrest

File: core-tests/src/test/java/org/mule/runtime/core/processor/chain/DefaultMessageProcessorChainTestCase.java [hunk 2]
```diff
@@ -612,7 +612,7 @@ public void testNestedInterceptingMPChainWithVoidReturn() throws MuleException,
                                                                 new ReturnVoidMPInterceptongMP(), new
                                 AppendingInterceptingMP("b")).build(),
                 new AppendingInterceptingMP("2"));
-        assertEquals("0before1before2after2after1", process(builder.build()
+        assertEquals("0before1beforeaafterabefore2after2after1", process(builder.build()
```
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/runtime/core/processor/chain/DefaultMessageProcessorChainTestCase.java]: hamcrest

File: core/src/main/java/org/mule/runtime/core/processor/AbstractInterceptingMessageProcessorBase.java [hunk 3]
```diff
@@ -90,19 +81,7 @@ else if (VoidMuleEvent.getInstance().equals(event))
             {
                 logger.trace("Invoking next MessageProcessor: '" + next.getClass().getName() + "' ");
             }
-
-            MessageProcessorExecutionTemplate executionTemplate = (!(next instanceof MessageProcessorChain)) ? messageProcessorExecutorWithNotifications : messageProcessorExecutorWithoutNotifications;
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/processor/AbstractInterceptingMessageProcessorBase.java]: where has this logic go?

File: core-tests/src/test/java/org/mule/runtime/core/processor/chain/DefaultMessageProcessorChainTestCase.java (hunk 1 above)
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/runtime/core/processor/chain/DefaultMessageProcessorChainTestCase.java]: where does the test value come from?

File: core-tests/src/test/java/org/mule/runtime/core/processor/chain/DefaultMessageProcessorChainTestCase.java (hunk 2 above)
Comment (marianogonzalez) [inline core-tests/src/test/java/org/mule/runtime/core/processor/chain/DefaultMessageProcessorChainTestCase.java]: value test?

File: core/src/main/java/org/mule/runtime/core/processor/AbstractInterceptingMessageProcessorBase.java (hunk 3 above)
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/processor/AbstractInterceptingMessageProcessorBase.java]: It's already in DefaultMessageProcessorChain.

**Pull request merged**